modules/Forum/pages/forum/view_topic.php

```diff
@@ -334,64 +334,41 @@
                     $available_hooks,
                 ));
 
-                // Alerts + Emails
-                $users_following = DB::getInstance()->get('topics_following', ['topic_id', $tid])->results();
-                if (count($users_following)) {
-                    $users_following_info = [];
-                    foreach ($users_following as $user_following) {
-                        if ($user_following->user_id != $user->data()->id) {
-                            if ($user_following->existing_alerts == 0) {
```

Comment:
You're no longer checking if user has already been notified of new activity since last visit

Can be very spammy if user get email/alert for every single new post

Reply:
resolved, added to where clause for fetching notification recipients